FishBunDemo/src/main/java/com/sangcomz/fishbundemo/ImageAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package com.sangcomz.fishbundemo
+
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.recyclerview.widget.RecyclerView
+import com.squareup.picasso.Picasso
+import java.util.*
+
+/**
+ * Created by sangc on 2015-11-06.
+ */
+class ImageAdapter(
+    private var imageController: ImageController,
+    private var imagePaths: ArrayList<Uri>
+) : RecyclerView.Adapter<ImageAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.item, parent, false)
+        return ViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val imagePath = imagePaths[position]
+        Picasso
+            .get()
+            .load(imagePath)
+            .fit()
+            .centerCrop()
+            .into(holder.imageView)
+        holder.imageView.setOnClickListener { imageController.setImgMain(imagePath) }
+    }
+
+    fun changePath(imagePaths: ArrayList<Uri>) {
+        this.imagePaths = imagePaths
+        imageController.setImgMain(imagePaths[0])
+        notifyDataSetChanged()
+    }
+
+    override fun getItemCount(): Int = imagePaths.size
+
+    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
I've been using inner keyword at `RecyclerView ViewHolder` since it looks more readable to me. But, I agree with you. We shouldn't use it if `Viewholder `doesn't access adapter's property.